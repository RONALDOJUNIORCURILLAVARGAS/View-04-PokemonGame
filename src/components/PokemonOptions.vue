<template>
  <div>
     <div class="options-container">
       <ul>
         <li v-for="pokemon in pokemons" 
         :key="pokemon.id"
         @click="$emit('selectionPokemon',pokemon.id)">
           {{pokemon.name}}
         </li>
         
       </ul>
     </div>
  </div>
</template>

<script>
export default {
  props:{
    pokemons:{
      type: Array,
      required:true
    }
  },
  methods: {
    hola(){
     // this.$emit()
    }
  },
  
}
</script>

<style scoped>
/* Pokemon Options */
ul {
    list-style-type: none;
    padding: 0px;
    
}
li {
    background-color: rgb(239, 83, 80);
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    margin-bottom: 10px;
    width: 250px;
    padding: 10px 20px;
    transition: all .3s;
    color: rgb(235, 231, 0);
    font-size: 24px;
    font-weight: 700;
    text-shadow: 2px 2px 0 rgb(0, 95, 204), -2px -2px 0 rgb(0, 95, 204), 2px -2px 0 rgb(0, 95, 204), -2px 2px 0 rgb(0, 95, 204);
}

li:hover {
    background-color: rgba(235, 231, 0,0.8);
    transform: scale(1.2);
    border: 2px solid rgb(239, 83, 80);
}

.options-container {
    display: flex;
    justify-content: center;
}
</style>