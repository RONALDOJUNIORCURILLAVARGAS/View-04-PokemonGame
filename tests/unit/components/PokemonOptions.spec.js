import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { pokemons } from "../mocks/pokemons.mock";
describe("PokemonOptions Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        //pokemons:pokemons
        pokemons,
      },
    })
  })
  test("debe de hacer match con el snapshot", () => {
    //console.log(wrapper.html())
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('debe de mostrar las 4 opciones correctamente', () => {
      //debe de tener 4 li
      //console.log(pokemons)
      const lista=wrapper.findAll('li')
      expect(lista.length).toBe(4);

      //cada li debe de tener su nombre del pokemon
      //expect(lista[0].text()).toBe('pikachu');
      for (const pokemon in pokemons){
        //console.log(pokemon)
        //console.log(pokemons[pokemon].name)
        expect(lista[pokemon].text()).toBe(pokemons[pokemon].name)
      }
      //Hacer esto es lo mismo
      expect(lista[0].text()).toBe('pikachu')
      expect(lista[1].text()).toBe('venusaur')
      expect(lista[2].text()).toBe('charmeleon')
      expect(lista[3].text()).toBe('mew')
  })
  test('debe de emitir "selection con sus respectivos parametros al hacer click', () => {
      const [li1,li2,li3,li4]=wrapper.findAll('li')
      //Utilizamos un disparador para simular un click
      li1.trigger('click')
      li2.trigger('click')
      li3.trigger('click')
      li4.trigger('click')

      console.log(wrapper.emitted('selectionPokemon'))
      expect(wrapper.emitted('selectionPokemon').length).toBe(4)
      expect(wrapper.emitted('selectionPokemon')[0]).toEqual([1])
      expect(wrapper.emitted('selectionPokemon')[1]).toEqual([3])
      expect(wrapper.emitted('selectionPokemon')[2]).toEqual([5])
      expect(wrapper.emitted('selectionPokemon')[3]).toEqual([15])
  })
  
  
})
