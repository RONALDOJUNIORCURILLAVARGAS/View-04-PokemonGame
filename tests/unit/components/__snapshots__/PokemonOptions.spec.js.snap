// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe de hacer match con el snapshot 1`] = `
<div>
  <div class="options-container">
    <ul>
      <li>pikachu</li>
      <li>venusaur</li>
      <li>charmeleon</li>
      <li>mew</li>
    </ul>
  </div>
</div>
`;
